### DISCUSSION TITLE
Azure AD groups or App Roles to Abp Permissions grants

### DISCUSSION THREAD

**Discussion by maniani-dev** [abpframework/abp#10522] [General]
How can I add Azure AD app roles to abp permission grants?

At the moment, I am able to get the roles for example Company.Task.Create and just want to know how I can add that into abp permission grants.

Not sure if this is a good way of doing it but works:

Az ad passes the app roles claims and all I'm doing below is inserting  them into permission grants
`CheckIdentityErrors(await UserManager.CreateAsync(user));

            if (info.Principal.HasClaim(c => c.Type == ClaimTypes.Role))
            {
                var identityClaims = info.Principal.FindAll(ClaimTypes.Role).ToList();

                if (identityClaims.Any())
                {
                    var availablePermissions = await _permissionManager.GetAllForUserAsync(user.Id);

                    foreach (var claim in identityClaims)
                    {
                        if(availablePermissions.Any(x=>x.Name == claim.Value))
                        {
                            await _permissionManager.SetForUserAsync(user.Id, claim.Value, true);
                        }
                    }
                }
            }`

Let me know what you think

**Comment by maliming** [comment]
See https://docs.abp.io/en/abp/latest/Authorization#advanced-topics
